src/security-rules/security-rules-api-client.ts

```diff
@@ -119,6 +124,38 @@ export class SecurityRulesApiClient {
       });
   }
 
+  public listRulesets(pageSize: number = 100, pageToken?: string): Promise<ListRulesetsResponse> {
+    if (!validator.isNumber(pageSize)) {
+      const err = new FirebaseSecurityRulesError('invalid-argument', 'Invalid page size.');
+      return Promise.reject(err);
+    }
+    if (pageSize < 1 || pageSize > 100) {
+      const err = new FirebaseSecurityRulesError(
+        'invalid-argument', 'Page size must be between 1 and 100.');
+      return Promise.reject(err);
+    }
+    if (typeof pageToken !== 'undefined' && !validator.isNonEmptyString(pageToken)) {
+      const err = new FirebaseSecurityRulesError(
+        'invalid-argument', 'Next page token must be a non-empty string.');
+      return Promise.reject(err);
+    }
+
+    const data = {
+      pageSize,
+      pageToken,
+    };
+    if (!pageToken) {
+      delete data.pageToken;
+    }
+
+    const request: HttpRequestConfig = {
+      method: 'GET',
+      url: `${this.url}/rulesets`,
+      data,
+    };
+    return this.sendRequest<ListRulesetsResponse>(request);
```

Comment:
consider providing a convenience method that repeatedly paginates until the end and then returns the entire list

Reply:
We haven't really done that sort of thing in Node.js before. But that's something we can certainly consider in the future. 

I think we haven't done this for APIs like [`auth.listUsers()`](https://firebase.google.com/docs/reference/admin/node/admin.auth.Auth.html#listusers), since a project can potentially have millions of users, and we don't want an implementation that buffers all that data in memory. But since there's a hard limit on how many rulesets a project can have, it is probably not a bad idea here.

In some of the other languages like Java and Python we already have established iterable/iterator patterns to support this use case cleanly. e.g. https://firebase.google.com/docs/reference/admin/java/reference/com/google/firebase/auth/ListUsersPage.html

Reply:
clever